src/Microsoft.AspNet.OData.Shared/ODataAPIHandler.cs

```diff
@@ -64,129 +66,203 @@ public abstract class ODataAPIHandler<TStructuralType>: IODataAPIHandler where T
         public abstract IODataAPIHandler GetNestedHandler(TStructuralType parent, string navigationPropertyName);
 
         /// <summary>
-        /// The parent object.
+        /// Apply handlers to the top level resource and the nested resources.
         /// </summary>
-        internal TStructuralType ParentObject { get; set; }
-
-        /// <summary>
-        /// Apply OdataId for a resource with OdataID container.
-        /// </summary>
-        /// <param name="resource">resource to apply odata id on.</param>
+        /// <param name="resource">Resource to execute.</param>
         /// <param name="model">The model.</param>
-        internal virtual void UpdateLinkedObjects(TStructuralType resource, IEdmModel model)
+        /// <param name="apiHandlerFactory">API Handler Factory.</param>
+        public virtual void ExecuteDeepInsert(TStructuralType resource, IEdmModel model, ODataAPIHandlerFactory apiHandlerFactory)
```

Comment:
I feel a certain type of way about `Execute` in `ExecuteDeepInsert`... Maybe use `HandleBulkInsert` or `HandleDeepInsert` since `Handle` is more widely used with the feature

Reply:
What if we used `DeepInsert`?